docs/source/mturk.rst

```diff
@@ -0,0 +1,181 @@
+2. Let all the agents in the world ``observe()`` and ``act()`` once, by calling ``parley()`` on the world.
+3. Ask the human Turker to rate the dialog model agent's response from 0-10.
+
+There are also two agents in this task: one is the human Turker, the other is the local Model Evaluator agent (herein called *evaluator agent*) running on the ParlAI user's machine. The purpose of evaluator agent is to initialize the dialog model and the world, get context and response from the dialog model by calling ``parley()`` on the world, and then ask for rating from the Turker.
+
+The flow of the task is hence determined by how ``observe()`` and ``act()`` are implemented in ``ModelEvaluatorAgent`` class in ``agents.py`` file. Note that since the evaluator agent only speaks once asking for Turker's rating, it doesn't need to use ``turn_index`` to keep track of the turns. 
+
+The flow of the task is as follows:
+
+Initialization:
+
+1. Evaluator agent is called ``__init__()``, which creates a world with a dialog model agent (``ir_baseline``) and a dataset (``MovieDD-Reddit``).
+
+At first turn:
+
+1. Evaluator agent is called ``act()``, which calls ``parley()`` once on the world, gets both the context and the dialog model's response, and returns a message asking the Turker to rate the response (with ``episode_done`` set to ``True``).
+2. Turker receives evaluator agent's prompt, and provides her rating.
+3. Evaluator agent is called ``observe()``, and receives Turker's rating.
+
+After one turn, the task is finished, and the Turker's work is submitted for your review.
+
+Creating Your Own Task
+---------------
+
+ParlAI provides a generic MTurk dialog interface that one can use to implement any kind of dialog tasks. To create your own task, start with reading the tutorials on the provided examples, and then copy and modify the example code to create your task. 
+
+A few things to keep in mind:
+
+1. Each conversation always starts with the local agent speaking first. (Its ``act()`` method is automatically called at the beginning of the conversation.)
+2. To end a conversation, you should set ``episode_done`` to ``True`` when returning a message from ``act()``, which means the agent expects ``observe()`` to be called next and then the conversation will end.
+3. You can provide a different context to each of the conversations (identified by ``self.conversation_id`` field), hence ensuring that the context that each Turker responds to is unique.
+4. Make sure to test your dialog task using MTurk's sandbox mode before pushing it live, by setting ``is_sandbox`` to ``True`` in ``run_mturk.py``.
+
+
+Running a Task
+---------------
+
+To run an MTurk task, first ensure that the task directory is in ``parlai/mturk/tasks/``. Then, set the following variables in ``run_mturk.py`` file:
+
+.. code-block:: python
+
+    task_module_name = 'parlai.mturk.tasks.your_task_name'
```

Comment:
this should use the '-t' flag instead of having users edit the file, like we do for regular parlai tasks